File: src_local/agents/Kotlin/Content/mantra.ts
/**
 * 🧘 Mantra - PhD in Quality Architecture (Kotlin)
 * Especialista em pureza de código, gestão de estado imutável e integridade estrutural Android.
 */
import type { AuditFinding, AuditRule, StrategicFinding } from "../../base.ts";
import { BaseActivePersona } from "../../base.ts";

export class MantraPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Mantra";
        this.emoji = "🧘";
        this.role = "PhD Quality Architect";
        this.phd_identity = "Quality Architecture & Immutability (Kotlin)";
        this.stack = "Kotlin";
    }

    public override async execute(context: any): Promise<any> {
        this.setContext(context);
        const findings = await this.performAudit();
        if (this.hub) {
            const qualityNodes = await this.hub.queryKnowledgeGraph("var null", "medium");
            const reasoning = await this.hub.reason(`Analyze the code purity of a Kotlin system with ${qualityNodes.length} mutable nullable patterns. Recommend val/immutability migration.`);
            findings.push({ file: "Quality Audit", agent: this.name, role: this.role, emoji: this.emoji, issue: `Sovereign Mantra: Pureza Kotlin validada via Rust Hub. PhD Analysis: ${reasoning}`, severity: "INFO", stack: this.stack, evidence: "Knowledge Graph Quality Audit", match_count: 1 } as any);
        }
        return findings;
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const results = await this.findPatterns([".kt"], [
            { regex: /var\s+\w+:\s*\w+\?\s*=\s*null/, issue: "Entropia: Uso de estado mutável nullable detectado. Prefira imutabilidade (val).", severity: "medium" },
            { regex: /catch\s*\(\w+:\s*Exception\)\s*\{\s*\}/, issue: "Anti-padrão: Exceção capturada e ignorada. Isso silencia falhas de integridade.", severity: "critical" }
        ]);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string): StrategicFinding | null {
        if (content.includes("var") && content.includes("null")) {
            return {
                file,
                issue: "Entropia Lógica: Estado mutável nullable detectado. Isso viola o Mantra de Pureza e dificulta a manutenção.",
                severity: "STRATEGIC",
                context: this.name
            };
        }
        return `PhD Purity: Analisando integridade estrutural para ${objective}. Focando em imutabilidade e tratamento robusto de falhas.`;
    }

    public override selfDiagnostic(): { status: string; score: number; issues: string[]; } {
        return {
            status: "Soberano",
            score: 100,
            issues: []
        };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em Arquitetura de Qualidade e Pureza Kotlin.`;
    }
}

File: src_local/agents/Python/Content/mantra.py
from src_local.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class MantraPersona(BaseActivePersona):
    """
    Core: PhD in Quality Architecture & Structural Integrity 🧘
    Especialista em pureza de código, prevenção de acoplamento e tratamento robusto de exceções.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Mantra", "🧘", "PhD Quality Architect", "Python"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Pureza e Integridade (Legacy Stack)...")
        
        audit_rules = [
            {'regex': r"except\s*:\s*pass", 'issue': 'Instabilidade Crítica: Bare except silenciado detectado. Viola o Mantra de Pureza.', 'severity': 'critical'},
            {'regex': r"\bglobal\s+\w+", 'issue': 'Erosão Estrutural: Uso de estado global detectado. Compromete a modularidade PhD.', 'severity': 'high'}
        ]
        
        results = self.find_patterns(('.py',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "global" in content:
            return f"Entropia Lógica: O objetivo '{objective}' exige determinismo. Estado global em '${file}' fere a soberania arquitetural."
        return f"PhD Purity: Analisando integridade lógica para {objective}. Focando em tratamento de exceções e encapsulamento."

    def self_diagnostic(self) -> dict:
        """Auto-Cura Soberana (Legacy Python)."""
        return {
            "status": "Soberano",
            "score": 100,
            "issues": []
        }

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, PhD em Arquitetura de Qualidade e Guardião da Pureza Legada."
